### PR TITLE
feat: port project_user_role_event to Go listener (#102)

### LINKED ISSUE #102: refactor: port `project_user_role_event` projector to Go event listener
## Scope

Port the `project_user_role_event(event events) RETURNS void` PL/pgSQL function to a Go event listener following the pattern from #77 (system actions) and #81 (search indexer).

## Function characterisation

tiny, two-row insert/delete on `user_roles_projection`.

## Current location

- PL/pgSQL function: `project_user_role_event` in `internal/store/migrations/` (find with `grep -l project_user_role_event internal/store/migrations/*.sql`)
- Trigger dispatch: `project_event()` → `project_user_role_event` for the matching stream type
- Event types projected include: `UserRoleAssigned`/`UserRoleRevoked`

## Target

- New Go file: `internal/projectors/user_role_projector.go` (or wherever the convention lands after #81)
- Implements `Apply(ctx, tx, event)` for every event type the PL/pgSQL function handled
- Registered in `cmd/control/main.go` next to the other Go projectors
- Live path: invoked synchronously by the post-commit listener, in the same DB transaction as the event append (read-your-writes consistency preserved)
- Rebuild path: invoked by the unified `RebuildAll` from #94

## Acceptance criteria

- [ ] Go projector handles every event type the PL/pgSQL function did
- [ ] Each event type covered by table-driven tests against testcontainers Postgres
- [ ] PL/pgSQL function deleted in a new migration; `project_event()` trigger no longer dispatches to it
- [ ] Existing handler tests still pass (read-your-writes preserved)
- [ ] CR review clean

## Tracking

Part of the Q2 2026 PL/pgSQL → Go listener migration (Phase 1) tracker.

### PR DESCRIPTION
**Pull request by PaulDotterer**
## Summary

Seventh port under the projector-migration pattern. Replaces `project_user_role_event` PL/pgSQL with `projectors.UserRoleListener`.

Two event types, single statement each — no transaction wrap needed:
- `UserRoleAssigned` — INSERT … ON CONFLICT (user_id, role_id) DO NOTHING preserves the PL/pgSQL projector's idempotency under reconciler replays.
- `UserRoleRevoked` — DELETE WHERE composite key — silently no-op on a miss (matches PL/pgSQL behaviour for repeated revoke events).

Composite-key fields (user_id, role_id) are explicitly validated at the listener layer so a malformed event can't write a row with empty keys.

## Read-your-writes preserved

`role_handler.AssignRoleToUser` checks `UserHasRole` before emitting Assigned, and that read sees the prior listener's write because `fireListeners` is synchronous (project memory `feedback_post_commit_listener_is_sync.md`).

## Test plan

Tests written FIRST per saved TDD feedback.

- [x] Pure decoders: happy paths, missing user_id/role_id, wrong stream/event type → `ErrIgnoredEvent`, malformed payloads
- [x] Integration: Assign→Revoke lifecycle, replay idempotency (3x Assign produces 1 row), Revoke-when-absent is a no-op, wrong-stream-type ignored
- [x] All tests pass locally (`go test ./server/internal/projectors/...`)

Refs tracker #107. Closes #102.

## Summary by CodeRabbit

* **Refactor**
  * Moved user-role event processing from database-side procedures into the application, centralizing projection handling and ensuring consistent role assignment/revocation behavior.

* **Tests**
  * Added end-to-end and unit tests for user role assign/revoke flows, covering idempotency, missing data, and edge cases to improve reliability.

### CODE REVIEW

**Comment by PaulDotterer** [comment]
@coderabbitai review

File: internal/store/migrations/023_user_role_projector_to_go.sql [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+-- +goose StatementBegin
+CREATE OR REPLACE FUNCTION project_user_role_event(event events) RETURNS void AS $$
+BEGIN
+    -- No-op: ported to projectors.UserRoleListener. See migration
+    -- comment + the listener wiring in cmd/control/main.go via
+    -- projectors.WireAll.
+    NULL;
+END;
+$$ LANGUAGE plpgsql;
+-- +goose StatementEnd
```
[the remark this answers is not in the document]
Comment (PaulDotterer) [reply internal/store/migrations/023_user_role_projector_to_go.sql:42]: Skipping the specific suggestion (add `user_role` to `AllRebuildTargets` + drop `rebuild_user_roles_projection`) — verified the orphan function doesn't exist in the repo (`grep -r rebuild_user_roles_projection server/` returns nothing; #94 dropped all 13 `rebuild_*_projection()` PL/pgSQL functions before this milestone started).

There IS a real bug here, just deeper than CR's framing: every entry in `AllRebuildTargets` whose stream type has been ported to a Go projector is now broken. `RebuildAll` dispatches via `SELECT project_<X>_event(events.*)` — once the migration replaces `project_<X>_event` with a no-op stub, the rebuild TRUNCATEs the projection and replays into the no-op, leaving the table empty.

Already broken right now (#101 merged):
- `roles` target → `project_role_event` (no-op) → wipes `roles_projection` on rebuild

Will break next ports if added to `AllRebuildTargets`:
- `user_role` (this PR) — currently NOT in the registry, so adding an entry pointing to the stub would actively make things worse
- `tokens` (#103) — already in registry → will break on merge
- `user_selections` (#106) — already in registry → will break on merge

The fix is broader: `runOneTarget` needs to dispatch through the Go listener pipeline (via `fireListeners` against synthesised `PersistedEvent`s) for ported stream types, not via the PL/pgSQL function. That's a separate tracker — not appropriate to bundle into a per-stream port PR.

Filing as a follow-up issue; will reference here once opened. For #102 specifically, NOT adding `user_role` to `AllRebuildTargets` keeps the projection consistent (it's only ever updated by the live event flow, never by emergency rebuild) — same as the security_alert / totp / lps_password / luks_key / server_settings ports already merged.

**Comment by PaulDotterer** [comment]
Filed the broader RebuildAll bug as #125. Per its rationale, NOT adding user_role to AllRebuildTargets in this PR — adding an entry pointing to the now-no-op project_user_role_event stub would actively wipe user_roles_projection on emergency rebuild.

**Comment by PaulDotterer** [comment]
@coderabbitai review
